tests/v3/test_codecs/test_endian.py

```diff
@@ -0,0 +1,87 @@
+from typing import Literal
+
+import numpy as np
+import pytest
+
+import zarr.v2
+from zarr.abc.store import Store
+from zarr.array import AsyncArray
+from zarr.buffer import default_buffer_prototype
+from zarr.codecs import BytesCodec
+from zarr.store.core import StorePath
+from zarr.testing.utils import assert_bytes_equal
+
+from .test_codecs import _AsyncArrayProxy
+
+
+@pytest.mark.parametrize("store", ("local", "memory"), indirect=["store"])
+@pytest.mark.parametrize("endian", ["big", "little"])
+async def test_endian(store: Store, endian: Literal["big", "little"]) -> None:
+    data = np.arange(0, 256, dtype="uint16").reshape((16, 16))
+    path = "endian"
+    spath = StorePath(store, path)
+    a = await AsyncArray.create(
+        spath,
+        shape=data.shape,
+        chunk_shape=(16, 16),
+        dtype=data.dtype,
+        fill_value=0,
+        chunk_key_encoding=("v2", "."),
+        codecs=[BytesCodec(endian=endian)],
+    )
+
+    await _AsyncArrayProxy(a)[:, :].set(data)
+    readback_data = await _AsyncArrayProxy(a)[:, :].get()
+    assert np.array_equal(data, readback_data)
+
+    # Compare with zarr-python
+    z = zarr.v2.create(
```

Comment:
We probably need to change these before we get rid of the v2 package.

Reply:
i changed "zarr-python" to "v2", so it looks a little less odd